src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -536,26 +546,35 @@ DirectStepI::qualities (
 {
     if (srcRedeems)
     {
-        return std::make_pair(
-            quality (
-                sb, src_, dst_, currency_,
-                false),
-            QUALITY_ONE);
+        if (!prevStep_)
+            return std::make_pair(QUALITY_ONE, QUALITY_ONE);
```

Comment:
If you want, I believe this can be shortened to `{QUALITY_ONE, QUALITY_ONE}`.  Similarly for the other two `make_pair`s in this method.  Your choice.